xonsh/procs/readers.py

```diff
@@ -54,31 +54,27 @@ def read_queue(self):
 
     def read(self, size=-1):
         """Reads bytes from the file."""
-        i = 0
-        buf = b""
-        while size < 0 or i != size:
+        buf = bytearray()
+        while size < 0 or len(buf) != size:
```

Comment:
I wonder if this shouldn't be `len(buf) <= size` instead and have the method return a truncated value.